packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -207,58 +296,31 @@ export class SitemapRequestList implements IRequestList {
         const url = this.reclaimed.values().next().value as string | undefined;
         if (url !== undefined) {
             this.reclaimed.delete(url);
-            return this.requestData.get(url)!;
```

Comment:
Not cool - this was actually useful. The reclaimed `Request` object can contain valuable data (e.g., retry count)